File: OOP-in-Java-Homework/src/_01_geometry/plane_shapes/Circle.java
package _01_geometry.plane_shapes;

import vertices.Vertex2D;

public class Circle extends PlaneShape {
	
	private double radius;
	
	public Circle(Vertex2D center, double radius) {
		super(new Vertex2D[] { center });
		this.setRadius(radius);
	}

	public double getRadius() {
		return radius;
	}

	public void setRadius(double radius) {
		if (radius < 0) {
			throw new IllegalArgumentException("Radius can't be negative!");
		}
		
		this.radius = radius;
	}
	
	@Override
	public double getArea() {
		return Math.PI * this.getRadius() * this.getRadius();
	}

	@Override
	public double getPerimeter() {
		return Math.PI * this.getRadius() * 2;
	}

	@Override
	public String toString() {
		return "Circle: \n" + this.getVertices()[0].toString() +
				"\nperimeter: " + this.getPerimeter() + "\narea:" + this.getArea();
	}
}

File: Encapsulation-and-Polymorphism-Homework/01. Shapes/Circle.cs
namespace _01.Shapes
{
    using System;

    public class Circle : IShape
    {
        private double radius;

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius
        {
            get { return this.radius; }
            set
            {
                if (value == null || value < 0)
                {
                    throw new ArgumentNullException("radius", "Radius can't be null or negative!");
                }
                this.radius = value;
            }
        }

        public double CalculateArea()
        {
            return Math.PI * this.Radius * this.Radius;
        }

        public double CalculateParameter()
        {
            return 2 * Math.PI * this.Radius;
        }
    }
}